src/nvim/normal.c

```diff
@@ -2145,23 +2147,39 @@ void do_check_scrollbind(bool check)
   old_leftcol = curwin->w_leftcol;
 }
 
+static int get_scrollbind_amount(win_T *orig_wp, int old_topline, int topfill_diff)
+{
+  int orig_topline = orig_wp->w_topline;
+
+  if (orig_topline == old_topline) {
+    return -topfill_diff;
+  }
+
+  int count = -abs(topfill_diff);
```

Comment:
This doesn't seems correct. I think it should be like this:
```suggestion
  int count = orig_topline > old_topline ? -topfill_diff : topfill_diff;
```